apollo-router-core/src/json_ext.rs

```diff
@@ -212,6 +132,160 @@ impl ValueExt for Value {
             (a, b) => a == b,
         }
     }
+
+    /// Insert a specific value at a subpath.
+    ///
+    /// This will create objects, arrays and null nodes as needed if they
+    /// are not present: the resulting Value is meant to be merged with an
+    /// existing one that contains those nodes.
+    #[track_caller]
+    fn from_path(path: &Path, value: Value) -> Value {
```

Comment:
it would make sense to add back an `insert_data` method following the same pattern, since in various places we create a `Value` to immediately merge it

Reply:
`Value::insert` was added in 9793f76